projects/packages/videopress/src/client/admin/components/admin-page/index.tsx

```diff
@@ -105,17 +109,61 @@ const Admin = () => {
 
 	const [ isSm ] = useBreakpointMatch( 'sm' );
 
+	const canDrop = hasVideoPressPurchase && isRegistered && hasVideos && ! loading;
+
+	const {
+		isDraggingOver,
+		handleDragOverEvent,
+		handleDragLeaveEvent,
+		handleDropEvent,
+		inputRef,
+		handleFileInputChangeEvent,
+	} = useDropFiles( {
+		canDrop,
+		onSelectFiles: handleFilesUpload,
+	} );
+
 	const addNewLabel = __( 'Add new video', 'jetpack-videopress-pkg' );
 	const addFirstLabel = __( 'Add your first video', 'jetpack-videopress-pkg' );
 	const addVideoLabel = hasVideos ? addNewLabel : addFirstLabel;
 
 	useAnalyticsTracks( { pageViewEventName: 'jetpack_videopress_admin_page_view' } );
 
+	useEffect( () => {
+		document.addEventListener( 'drop', handleDropEvent );
+		document.addEventListener( 'dragover', handleDragOverEvent );
+		document.addEventListener( 'dragleave', handleDragLeaveEvent );
+
+		return () => {
+			document.removeEventListener( 'drop', handleDropEvent );
+			document.removeEventListener( 'dragover', handleDragOverEvent );
+			document.removeEventListener( 'dragleave', handleDragLeaveEvent );
+		};
+	}, [ handleDropEvent, handleDragOverEvent, handleDragLeaveEvent ] );
```

Comment:
have you considered moving this logic into the hook?

Reply:
I did. The problem here is that while the `VideoUploadArea` component controls its own template, the whole-page drop-zone needs to do it with `addEventListener` instead of, say, `onDrop={ handleDropEvent }`, so this looks like a use case that is specific to the admin page.
It can be moved there if we add a `element` property to the hook. What do you think?

Reply:
Done.